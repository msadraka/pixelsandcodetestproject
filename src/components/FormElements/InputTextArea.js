import React from 'react';
import { Input } from 'antd';
import type { FormProps } from 'redux-form';

import StyleWrapper from './input.style';

type Props = {
  className: string,
  label: string
} & FormProps;

export default function(props: Props) {
  const {
    input,
    className,
    label,
    placeholder,
    meta: { touched, error },
    afterChange,
    ...reset
  } = props;

  function handleBlur(event: Object) {
    input.onBlur();
    const value = event.target.value.trim();
    input.onChange(value);
    if (afterChange) afterChange(value);
  }

  function handleChange(event: Object) {
    input.onChange(event);
    if (afterChange) {
      afterChange(event.target.value);
    }
  }

  return (
    <StyleWrapper>
      <div
        className={`c-input textarea ${className} ${
          touched && error ? ' hasError' : ''
        }`}
      >
        <label htmlFor={input.name}>
          {label && <span className="labelText">{label}</span>}

          <Input.TextArea
            {...input}
            {...reset}
            id={input.name}
            className="roundInput textareaInput"
            rows={5}
            style={{ width: '100%' }}
            placeholder={placeholder}
            onBlur={handleBlur}
            onChange={handleChange}
          />
        </label>
        {touched && error && <span className="text-danger">{error}</span>}
      </div>
    </StyleWrapper>
  );
}
