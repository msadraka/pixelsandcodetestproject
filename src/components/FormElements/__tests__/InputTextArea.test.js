import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { get } from 'lodash';
import InputTextArea from '../InputTextArea';

global.Input = require('antd').Input;

describe('<InputTextArea />', () => {
  const defaultProps = {
    input: { name: 'field' },
    label: 'field',
    meta: { touched: false, error: null }
  };
  const BaseFieldLayout = props => (
    <InputTextArea {...defaultProps} {...props} />
  );
  // 1) snapshot
  it('take snapshot of <InputTextArea />', () => {
    const BaseFieldLayoutComponent = renderer
      .create(<BaseFieldLayout />)
      .toJSON();
    expect(BaseFieldLayoutComponent).toMatchSnapshot();
  });

  // 2) testing props
  it('check  className props', () => {
    const props = {
      className: 'class',
      label: 'label'
    };
    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    expect(
      BaseFieldLayoutComponent.find('.textarea').hasClass('class')
    ).toBeTruthy();

    expect(
      BaseFieldLayoutComponent.find('span').hasClass('labelText')
    ).toBeTruthy();
  });

  // 3) testing event
  it('checking onBlur event component', () => {
    const props = {
      afterChange: jest.fn(),
      input: {
        value: '',
        name: 'field',
        onBlur: jest.fn(),
        onChange: value => Object.assign(props.input, { value })
      }
    };

    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    BaseFieldLayoutComponent.find('#field')
      .at(0)
      .simulate('blur', {
        target: { value: ' SomeText ' }
      });

    expect(get(props, 'input.value', '')).toBe('SomeText');
  });

  it('checking onBlur event component', () => {
    const props = {
      afterChange: jest.fn(),
      input: {
        value: '',
        name: 'field',
        onBlur: jest.fn(),
        onChange: value => Object.assign(props.input, { value })
      }
    };

    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    BaseFieldLayoutComponent.find('#field')
      .at(0)
      .simulate('blur', {
        target: { value: ' SomeText ' }
      });

    expect(get(props, 'input.value', '')).toBe('SomeText');
  });

  it('checking onBlur event component without afterChange', () => {
    const props = {
      input: {
        value: '',
        name: 'field',
        onBlur: jest.fn(),
        onChange: value => Object.assign(props.input, { value })
      }
    };

    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    BaseFieldLayoutComponent.find('#field')
      .at(0)
      .simulate('blur', {
        target: { value: ' SomeText ' }
      });

    expect(get(props, 'input.value', '')).toBe('SomeText');
  });
  it('checking onChange event component', () => {
    const props = {
      afterChange: jest.fn(),
      input: {
        value: '',
        name: 'field',
        onBlur: jest.fn(),
        onChange: value => Object.assign(props.input, { value })
      }
    };

    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    BaseFieldLayoutComponent.find('#field')
      .at(0)
      .simulate('change', {
        target: { value: 'SomeText' }
      });

    expect(get(props, 'input.value.target.value', '')).toBe('SomeText');
  });
  it('checking onChange event component without afterChange props', () => {
    const props = {
      input: {
        value: '',
        name: 'field',
        onBlur: jest.fn(),
        onChange: value => Object.assign(props.input, { value })
      }
    };

    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    BaseFieldLayoutComponent.find('#field')
      .at(0)
      .simulate('change', {
        target: { value: 'SomeText' }
      });

    expect(get(props, 'input.value.target.value', '')).toBe('SomeText');
  });

  // 4) test condition
  it('testing error of component', () => {
    const props = {
      input: { name: 'field' },
      meta: { touched: true, error: 'Required' }
    };
    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);

    expect(
      BaseFieldLayoutComponent.find('div')
        .at(1)
        .hasClass('hasError')
    ).toBeTruthy();

    expect(
      BaseFieldLayoutComponent.find('span')
        .at(1)
        .text()
    ).toEqual('Required');
  });

  it('checking Labeled component', () => {
    const props = {
      input: { name: 'field' },
      label: 'Field'
    };
    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);
    expect(BaseFieldLayoutComponent.find('span').text()).toEqual('Field');
  });

  // 5) checking Props types
  it('checking props types', () => {
    const props = {
      className: 'hasClass',
      label: 'label'
    };
    const BaseFieldLayoutComponent = mount(<BaseFieldLayout {...props} />);
    expect(BaseFieldLayoutComponent.prop('className')).toBeString();
    expect(BaseFieldLayoutComponent.prop('label')).toBeString();
  });
});
