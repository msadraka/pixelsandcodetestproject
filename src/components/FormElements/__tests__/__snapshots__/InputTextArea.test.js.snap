// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextArea /> take snapshot of <InputTextArea /> 1`] = `
<div
  className="inputstyle___default-cujm1d-0 ecaWej"
>
  <div
    className="c-input textarea undefined "
  >
    <label
      htmlFor="field"
    >
      <span
        className="labelText"
      >
        field
      </span>
      <textarea
        className="ant-input roundInput textareaInput"
        id="field"
        name="field"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        rows={5}
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </label>
  </div>
</div>
`;
